tiledb/sm/buffer/buffer_list.cc

```diff
@@ -52,7 +52,7 @@ Status BufferList::add_buffer(Buffer&& buffer) {
   return Status::Ok();
 }
 
-Status BufferList::get_buffer(uint64_t index, Buffer** buffer) {
+Status BufferList::get_buffer(uint64_t index, const Buffer** buffer) const {
```

Comment:
I'm not immediately seeing the reason for this `const` addition. Is it just a "that should have been const before so adding it now" type change, or was there something else that motivated it?

Reply:
Right. The C API has a `const tiledb_buffer_list_t*` parameter. Migrating to Handle use exposed the need for a `const Buffer**` argument in the internal API.

Reply:
Gotcha. Thanks!